### PR TITLE
[WiP] Sketch of the graph evaluation code

### PR DESCRIPTION
**Pull request by tpetricek**

### CODE REVIEW

File: client/src/languagePlugins/javascript/javascriptPlugin.ts [hunk 1]
```diff
@@ -209,6 +214,32 @@ class JavascriptBlockKind implements Langs.Block {
   export const javascriptLanguagePlugin : Langs.LanguagePlugin = {
     language: "javascript",
     editor: javascriptEditor,
+    evaluate: (scopeDictionary:{}, node:Graph.Node) => {
```
Comment (tpetricek) [inline client/src/languagePlugins/javascript/javascriptPlugin.ts]: We should not need to pass `scopeDictionary` to the `evaluate` function. Instead, we can get the variable values from the `JsCodeNode`.

File: client/src/languagePlugins/javascript/javascriptPlugin.ts [hunk 2]
```diff
@@ -209,6 +214,32 @@ class JavascriptBlockKind implements Langs.Block {
...
+          for (var e = 0; e < jsnode.exportedVariables.length; e++) {
+            returnArgs= returnArgs.concat(jsnode.exportedVariables[e]+":"+jsnode.exportedVariables[e]+",");
+          }
+          returnArgs = returnArgs.concat("}")
+          // console.log(returnArgs)
+          let evalCode = "function calc() {\n\t "+ jsnode.source +"\n\t return "+returnArgs+"\n}\n; calc()"
+          value = eval(evalCode);
+          break;
+        case 'export':
+          value = scopeDictionary[jsnode.variableName]
```
Comment (tpetricek) [inline client/src/languagePlugins/javascript/javascriptPlugin.ts]: Here, we should be able to access `jsnode.code` which should give us a `JsCodeNode` representing the value. The `value` of this node will then be the dictionary that we construct in the function passed to `eval` (the type of it is `any` so we may need to cast this to a dictionary).

File: client/src/languagePlugins/javascript/javascriptPlugin.ts (hunk 1 above)
Comment (myyong) [reply client/src/languagePlugins/javascript/javascriptPlugin.ts]: Have removed scope dictionary.

File: client/src/languagePlugins/javascript/javascriptPlugin.ts (hunk 2 above)
Comment (myyong) [reply client/src/languagePlugins/javascript/javascriptPlugin.ts]: Would you please check out the new version?

**Pull request merged**